pytorch_lightning/utilities/auto_restart.py

```diff
@@ -400,28 +401,31 @@ def _cycle_to_next_worker_and_reset(dataloader: DataLoader, state_dict: Dict[str
                 next(iter_dataloader._worker_queue_idx_cycle)
 
         # we can finally call reset and apply prefecthing.
-        iter_dataloader._reset = iter_dataloader._original_reset
+        setattr(iter_dataloader, "_reset", iter_dataloader._original_reset)
         iter_dataloader._reset(dataloader, first_iter=True)
     # return the iterator
     return iter_dataloader
 
 
 def _dataloader_to_state_dict(
     dataloader: DataLoader, iterator: Iterator, num_batches_processed: int = None
-) -> List[Dict[str, Any]]:
+) -> Dict[Union[int, str], Union[Dict[str, int], Optional[int]]]:
     """Convert a dataloader to its associated state dict."""
-    out = {}
+    out: Dict[Union[int, str], Union[Dict[str, int], Optional[int]]] = {}
     if iterator is not None:
-        out.update(_find_current_worker(iterator))
+        for iterator_k, iterator_v in _find_current_worker(iterator).items():
+            out[iterator_k] = iterator_v
 
     if not isinstance(dataloader.dataset, CaptureIterableDataset):
         fast_forward_sampler = _find_fast_forward_samplers(dataloader)
         if fast_forward_sampler is not None:
-            out.update(fast_forward_sampler.state_dict(num_batches_processed=num_batches_processed))
```

Comment:
Won't update works ?

Reply:
It returns me the following mypy error on my local:
```
pytorch_lightning/utilities/auto_restart.py:421: error: Argument 1 to "update" of "dict" has incompatible type "Dict[int, Dict[str, int]]"; expected "Mapping[Union[int, str], Union[Dict[str, int], int, None]]"  [arg-type]
```